### PR TITLE
Upgrade Anvil and use the built-in mechanism to sync sources. 

### PR DESCRIPTION
**Pull request by vRallev**
This feature requires a newer version of AGP, which itself requires a newer version of Gradle. Both dependencies have been upgraded as well.

### CODE REVIEW

**Review by changusmc — APPROVED** [review]

**Comment by changusmc** [comment]
> This feature requires a newer version of AGP, which itself requires a newer version of Gradle. Both dependencies have been upgraded as well.

Thanks for the AGP and Gradle bump, as well as the updated Anvil feature!

File: scoping/api/scoping.api [hunk 1]
```diff
@@ -119,8 +119,6 @@ public final class com/dropbox/kaiken/scoping/BuildConfig {
 	public static final field BUILD_TYPE Ljava/lang/String;
 	public static final field DEBUG Z
 	public static final field LIBRARY_PACKAGE_NAME Ljava/lang/String;
-	public static final field VERSION_CODE I
-	public static final field VERSION_NAME Ljava/lang/String;
```
Comment (vRallev) [inline scoping/api/scoping.api:123]: @changusmc I had to remove these. That was a change with AGP 4.1, see https://issuetracker.google.com/issues/154275579#comment3 Please take another look.

Comment (vRallev) [reply scoping/api/scoping.api:123]: If you prefer, then I can bring back these values for each module. But to me it seems unnecessary.

Comment (changusmc) [reply scoping/api/scoping.api:123]: I think it's okay, I doubt people are using these APIs

**Comment by vRallev** [comment]
I honestly don't know why this is:
```
debugAndroidTest > : No compatible devices connected.[TestRunner] FAILED 
Found 1 connected device(s), 0 of which were compatible.
```
I'm not too familiar with your setup, but I think the recommendation is to use a MacOS machine for instrumentation tests: https://github.com/ReactiveCircus/android-emulator-runner/issues/62 Other than that I have no idea. I can run tests locally fine.

**Comment by changusmc** [comment]
> I honestly don't know why this is:
> 
> ```
> debugAndroidTest > : No compatible devices connected.[TestRunner] FAILED 
> Found 1 connected device(s), 0 of which were compatible.
> ```
> 
> I'm not too familiar with your setup, but I think the recommendation is to use a MacOS machine for instrumentation tests: [ReactiveCircus/android-emulator-runner#62](https://github.com/ReactiveCircus/android-emulator-runner/issues/62) Other than that I have no idea. I can run tests locally fine.

We started seeing this recently, so it's not due to your changes. I'll verify locally and merge this.

**Pull request merged**